File: lkmc/add.c
#include <assert.h>

int main(void) {
    int i, j, k;
    i = 1;
    /* test-gdb-op1 */
    j = 2;
    /* test-gdb-op2 */
    k = i + j;
    /* test-gdb-result */
    if (k != 3)
        assert(0);
}

File: userland/gdb_tests/add.py
def test(self):
    self.sendline('tbreak main')
    self.sendline('continue')
    self.continue_to('op1')
    assert self.get_int('i') == 1
    self.continue_to('op2')
    assert self.get_int('j') == 2
    self.continue_to('result')
    assert self.get_int('k') == 3
